### PR TITLE
Parsing `export default ...` peeks to not overconsume tokens

### PR DESCRIPTION
**Pull request by Attamusc**
Fixes https://code.google.com/p/esprima/issues/detail?id=631

Paired with @mikesherov on this today.

Updates to the tests in `harmonymoduletest` to account for token generation is forthcoming (Will squash commits once those are in).

### CODE REVIEW

File: esprima.js [hunk 1]
```diff
@@ -3669,17 +3669,11 @@
 
     function parseExportDeclaration() {
         var backtrackToken, id, declaration = null,
```
Comment (mikesherov) [inline esprima.js]: isn't `bracktrackToken` no longer used?

File: esprima.js [hunk 2]
```diff
@@ -3669,17 +3669,11 @@
...
+            peekedToken, sourceElement,
```
Comment (mikesherov) [inline esprima.js]: maybe we should rename `peekedToken` to `possibleIdentifierToken`?

File: esprima.js (hunk 1 above)
Comment (Attamusc) [inline esprima.js]: Yup, not used at all.

File: esprima.js (hunk 2 above)
Comment (Attamusc) [inline esprima.js]: I like that better, since now it's just used to check if it's an identifier and not for potential rewinding.

**Comment by michaelficarra** [comment]
Please squash.

**Comment by mikesherov** [comment]
@michaelficarra yeah, we'll squash once tests land at the end. Thanks!

**Comment by Attamusc** [comment]
@michaelficarra Sure thing! Should have mentioned I'd do that at in the description.

**Comment by Attamusc** [comment]
Test changes are in. Since we modified code that touches the `default` branch of `parseExportDeclaration`, I added tokens to all `export default` tests.

**Comment by mikesherov** [comment]
@jeffmo or @ariya can someone please review this when you have some free time?

**Comment by jeffmo** [comment]
I'll take a look

-Jeff

File: test/harmonymodulestest.js [hunk 3]
```diff
@@ -34,66 +34,203 @@ var modulesTestFixture = {
...
         // default exports
         'export default 42;': {
-            type: 'ExportDeclaration',
-            'default': true,
-            declaration: {
-                type: 'Literal',
-                value: 42,
-                raw: '42',
-                range: [15, 17],
+            type: 'Program',
```
Comment (jeffmo) [inline test/harmonymodulestest.js]: Any particular reason you're starting at the root in these tests? It seems the test runner will allow it, but the rest of the tests in this fixture only include the first statement in the program body...so unless there's a good reason it'd be nice to retain consistency.

Comment (jeffmo) [inline test/harmonymodulestest.js]: Oh, I missed your earlier comment here: https://github.com/jquery/esprima/pull/310#issuecomment-73166939

**Comment by jeffmo** [comment]
This looks great, thanks for the PR!

**Comment by Attamusc** [comment]
@jeffmo Ya the inconsistency in that file now is kind of a bummer, but I plan on adding tokens to the rest of the cases in `harmonymoduletest` later in another PR to make sure no other issues like this pop up.

**Pull request merged**